param.go

```diff
@@ -113,22 +113,28 @@ func (f paramVisitorFunc) Visit(p param) paramVisitor {
 // visitor.
 //
 // This is very similar to how go/ast.Walk works.
-func walkParam(p param, v paramVisitor) {
+//
+// paramVisitor can end the walk early by returning nil.
+func walkParam(p param, v paramVisitor) (cont bool) {
 	v = v.Visit(p)
 	if v == nil {
-		return
+		return false
```

Comment:
Actually, this doesn't seem right. The idea behind `v.Visit` returning nil is
that it doesn't want to recurse further into that node. So if your visitor
doesn't care about *this* paramObject, it can return nil to indicate that we
shouldn't recurse into it, but we should still check the other objects. This
will end the whole iteration early.

If we want visitors to be able to abort the iteration, we can theoretically
make them return an `error` in addition to the "next" visitor.